tests/test_cuda_targeted_insertion_mover.py

```diff
@@ -25,13 +26,25 @@
 from timemachine.potentials import HarmonicBond, Nonbonded, SummedPotential
 
 
-def compute_ref_raw_log_prob(ref_exchange, water_idx, vi_mols, vj_mols, vol_i, vol_j, coords, box, new_coords):
+def compute_ref_raw_log_prob(
+    ref_exchange, water_idx, vi_mols, vj_mols, vol_i, vol_j, coords, box, sampled_mol_idx, new_coords
+):
     """Modified from timemachine.md.exchange.exchange_mover.TIBDExchangeMove.swap_vi_into_vj
     to support not resampling"""
     coords = coords.copy()
     log_weights_before_full = ref_exchange.batch_log_weights(coords, box)
     log_weights_before = log_weights_before_full[vi_mols]
 
+    # Verify that the probability of moving the specific water is not highly unlikely
+    src_idx = np.argwhere(vi_mols == sampled_mol_idx).reshape(-1)
+    log_probs_before = log_weights_before - logsumexp(log_weights_before)
+    probs_before = np.exp(log_probs_before)
+    median_probability = np.median(probs_before)
```

Comment:
Mostly as an interesting aside, but as you run the MC longer and longer the median probability increases. Which makes sense, as you try to minimize all of the water's positions.